src/api/app/controllers/concerns/webui/rescue_handler.rb

```diff
@@ -14,7 +14,9 @@ module Webui::RescueHandler
                       when 'create_branch?' then 'create_branch'
                       else exception.try(:query)
                       end
-      message = if pundit_action && exception.record
+      message = if exception.reason
```

Comment:
In order to keep the current flash message I propose to use the [latest version of Pundit exception](https://github.com/varvet/pundit/blob/master/lib/pundit.rb#L24-L41) and pass a `reason`.

Sadly `Pundit` didn't publish a newer release yet so I had to patch `NotAuthorizedError ` in order to achieve this.